controller/main.go

```diff
@@ -71,38 +71,26 @@ func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _
 	svc := svcRo.DeepCopy()
 	successRes := controllers.SyncStateSuccess
 	wasAllocated := c.isServiceAllocated(name)
-	if !c.convergeBalancer(l, name, svc) {
-		return controllers.SyncStateError
-	}
+	c.convergeBalancer(l, name, svc)
+
 	if wasAllocated && !c.isServiceAllocated(name) { // convergeBalancer may deallocate our service and this means it did it.
 		// if the service was deallocated, it may have have left room
 		// for another one, so we reprocess
 		level.Info(l).Log("event", "serviceUpdated", "msg", "removed loadbalancer from service, services will be reprocessed")
 		successRes = controllers.SyncStateReprocessAll
 	}
-	if reflect.DeepEqual(svcRo, svc) {
-		level.Debug(l).Log("event", "noChange", "msg", "service converged, no change")
-		return successRes
-	}
-
-	toWrite := svcRo.DeepCopy()
```

Comment:
why this logic is removed? This version was defensive and made sure that we are only updating the status part of the service. The new version on the other hand depends on the behaviour of convergeBalancer, making this a bit more fragile.

Reply:
Because it's cleaner in my opinion. Do we need to add this kind of protection of our own code?

Reply:
It's definitely cleaner. What I am asking here is if you are 100% sure we don't need it and you checked it.

If you did, then it's fine (despite it should still belong to a different commit and not to the "remove return value" one). If not, then please check before :-) I really don't have the answer because this part has more or less been stable and I thought that being defensive was fine.

Reply:
I'm sure. I'm also relying on the ci (assuming the failing test is not related). Nevertheless, you're right. This is out of the scope of this PR. Will remove it and create a new one.